netann/chan_status_manager.go

```diff
@@ -378,15 +416,8 @@ func (m *ChanStatusManager) processDisableRequest(outpoint wire.OutPoint,
 		return err
 	}
 
-	switch curState.Status {
-
-	// Channel is already disabled, nothing to do.
-	case ChanStatusDisabled:
```

Comment:
The entry *was* getting cleaned up for a cooperative close, since the channel would've started out in `ChanStatusEnabled` so we don't hit this case.

However, it was *not* getting cleaned up if the channel was already in `ChanStatusDisabled` (e.g. peer went offline for a while) and you do a force close.

More specifically (with extra debug logging for removing the channel from the map):

### Before this PR

cooperative close:

	[INF] NANN: Announcing channel(...) disabled [requested]
	[DBG] NANN: Removing channel(...) from map

peer goes down + force close disabled channel:

	[INF] NANN: Announcing channel(...) disabled [detected]
	>>> force close doesn't remove channel from map

### After this PR

cooperative close:

	[INF] NANN: Announcing channel(...) disabled [requested]
	[DBG] NANN: Removing channel(...) from map

peer goes down + force close disabled channel:

	[INF] NANN: Announcing channel(...) disabled [detected]
	[DBG] NANN: Removing channel(...) from map